CNV-40007: Added AAQ Operator docs (#83412)
openshift/openshift-docs: modules/virt-configuring-aaq-operator.adoc

      "vmiCalcConfigName": "DedicatedVirtualResources",
      "namespaceSelector": {
        "matchLabels": {
          "app": "my-app"
        }
      },
      "allowApplicationAwareClusterResourceQuota": true
    }
  }
}'
----
+
where:

`vmiCalcConfigName`:: Specifies how resource counting is managed for pods that run virtual machine (VM) workloads. Possible values are:
+
--
* `VmiPodUsage`: Counts compute resources for pods associated with VMs in the same way as native resource quotas and excludes migration-related resources.
* `VirtualResources`: Counts compute resources based on the VM specifications, using the VM RAM size for memory and virtual CPUs for processing.
* `DedicatedVirtualResources` (default): Similar to `VirtualResources`, but separates resource tracking for pods associated with VMs by adding a `/vmi` suffix to CPU and memory resource names. For example, `requests.cpu/vmi` and `requests.memory/vmi`.
--
`namespaceSelector`:: Determines the namespaces for which an AAQ scheduling gate is added to pods when they are created. If a namespace selector is not defined, the AAQ Operator targets namespaces with the `application-aware-quota/enable-gating` label as default.

> `allowApplicationAwareClusterResourceQuota`:: If set to `true`, you can create and manage the `ApplicationAwareClusterResourceQuota` object. Setting this attribute to `true` can increase scheduling time.

Reviewer: ```suggestion
`allowApplicationAwareClusterResourceQuota`: If set to `true`, you can create and manage the `ApplicationAwareClusterResourceQuota` object. Setting this attribute to `true` can increase scheduling time.
```

Author: The `::` is necessary for asciidoc description list syntax

// Module included in the following assemblies:
//
// * virt/virtual_machines/advanced_vm_management/virt-understanding-aaq-operator.adoc                

:_mod-docs-content-type: PROCEDURE                                    
[id="virt-configuring-aaq-operator_{context}"]                                   
= Configuring the AAQ Operator by using the CLI

You can configure the AAQ Operator by specifying the fields of the `spec.applicationAwareConfig` object in the `HyperConverged` custom resource (CR).

.Prerequisites
* You have access to the cluster as a user with `cluster-admin` privileges.
* You have installed the OpenShift CLI (`oc`).

.Procedure
* Update the `HyperConverged` CR by running the following command:
+
[source,terminal]
----
$ oc patch hco kubevirt-hyperconverged -n openshift-cnv --type merge -p '{
  "spec": {
    "applicationAwareConfig": {
      "vmiCalcConfigName": "DedicatedVirtualResources",
      "namespaceSelector": {
        "matchLabels": {
          "app": "my-app"
        }
      },
      "allowApplicationAwareClusterResourceQuota": true
    }
  }
}'
----
+
where:

> 

Reviewer 2: ```suggestion
```

`vmiCalcConfigName`:: Specifies how resource counting is managed for pods that run virtual machine (VM) workloads. Possible values are:
+
--

> * `VmiPodUsage` - Counts compute resources for pods associated with VMs in the same way as native resource quotas and excludes migration-related resources.

Reviewer 2: Consider colons rather than dashes here.

Reviewer 2: ISG asks that lists of sentence fragments skip terminating punctuation. 

(For translation's sake, I'd rewrite these as complete sentences and keep the punctuation.)

Author: Will replace dashes with colons but I think the sentence formatting is in alignment with the examples in the [contribution guide](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks).

* `VirtualResources` - Counts compute resources based on the VM specifications, using the VM RAM size for memory and virtual CPUs for processing.

> * `DedicatedVirtualResources` (default) - Similar to `VirtualResources`, but separates resource tracking for pods associated with VMs by adding a `/vmi` suffix to CPU and memory resource names. For example, `requests.cpu/vmi`, and `limits.memory/vmi`. This distinguishes them from other pods.

Reviewer: Not sure if we want to add it but As a best practice, it's recommended to use only requests or limits for VMIs, as they are counted the same, unlike with "regular" pods.

@dshchedr WDYT?

Reviewer 3: yeah, I guess we should only have `requests.cpu/vmi` here

> * `DedicatedVirtualResources` (default) - Similar to `VirtualResources`, but separates resource tracking for pods associated with VMs by adding a `/vmi` suffix to CPU and memory resource names. For example, `requests.cpu/vmi`, and `limits.memory/vmi`. This distinguishes them from other pods.

Reviewer 2: "This [noun] distinguishes them from other pods."

--

> `namespaceSelector`:: Determines the namespaces for which an AAQ scheduling gate is added to pods when they are created.

Reviewer: Please mention that the default selector is for namespaces with this label:

   application-aware-quota/enable-gating

> `allowApplicationAwareClusterResourceQuota`:: If set to `true`, you can create and manage the `ApplicationAwareClusterResourceQuota` object. Setting this attribute to `true` can cause some performance issues.

Reviewer: ```suggestion
`allowApplicationAwareClusterResourceQuota`:: If set to `true`, you can create and manage the `ApplicationAwareClusterResourceQuota` object. Setting this attribute to `true` can increase scheduling time.
```
